src/base/DeltaResolver.sol

```diff
@@ -59,4 +62,24 @@ abstract contract DeltaResolver is ImmutableState {
         if (_amount < 0) revert IncorrectUseOfTake();
         amount = uint256(_amount);
     }
+
+    /// @notice Calculates the amount for a settle action
+    function _mapSettleAmount(uint256 amount, Currency currency) internal view returns (uint256) {
+        if (amount == Constants.CONTRACT_BALANCE) {
+            return currency.balanceOfSelf();
+        } else if (amount == Constants.OPEN_DELTA) {
+            return _getFullSettleAmount(currency);
+        }
+        return amount;
+    }
+
+    /// @notice Calculates the amount for a swap action
+    function _mapSwapAmount(uint128 amount, Currency currency) internal view returns (uint128) {
```

Comment:
nit: Since I believe we will use this function we can use a more neutral name like mapInputAmount